packages/client-node/__tests__/tls/tls.test.ts

```diff
@@ -135,5 +137,36 @@ describe('[Node.js] TLS connection', () => {
       })
       expect(await resultSet.text()).toEqual('0\n1\n2\n')
     })
+
+    describe('Custom HTTPS agent', () => {
+      let httpRequestStub: jasmine.Spy<typeof Http.request>
+      beforeEach(() => {
+        httpRequestStub = spyOn(Http, 'request').and.callThrough()
+      })
+
+      it('should work with a custom HTTPS agent', async () => {
+        const agent = new https.Agent({
+          maxFreeSockets: 5,
+          ca: ca_cert,
+        })
+        const client = createClient({
+          url: 'https://server.clickhouseconnect.test:8443',
```

Comment:
can't we run the test against Cloud?

Reply:
We can, but this is slightly more annoying - it will have to choose between HTTP and HTTPS agent, plus I wanted to test the auth with a CA cert (so I'll need to add that to the repo as well).